stdlib/public/core/Runtime.swift

```diff
@@ -350,14 +350,20 @@ internal struct _Buffer72 {
 }
 
 #if !((os(macOS) || targetEnvironment(macCatalyst)) && arch(x86_64))
+#if arch(wasm32)
 // Note that this takes a Float32 argument instead of Float16, because clang
 // doesn't have _Float16 on all platforms yet.
+typealias _Float16Storage = Float32
```

Comment:
Naming here isn't really great either; this isn't really about "storage", but rather promoting to a type that's legal (in C) for the target.